src/Microsoft.OData.Core/Json/BufferingJsonReader.cs

```diff
@@ -999,6 +1350,589 @@ private void RemoveFirstNodeInBuffer()
+                            ODataJsonLightReaderUtils.ErrorPropertyBitMask.Code))
+                        {
+                            return Tuple.Create(false, error);
+                        }
+
+                        readErrorStringPropertyResult = await this.TryReadErrorStringPropertyValueAsync()
+                            .ConfigureAwait(false);
+                        if (!readErrorStringPropertyResult.Item1)
+                        {
+                            return Tuple.Create(false, error);
+                        }
+
+                        error.ErrorCode = readErrorStringPropertyResult.Item2;
+                        break;
+
+                    case JsonConstants.ODataErrorMessageName:
+                        if (!ODataJsonLightReaderUtils.ErrorPropertyNotFound(
+                            ref propertiesFoundBitmask,
+                            ODataJsonLightReaderUtils.ErrorPropertyBitMask.Message))
+                        {
+                            return Tuple.Create(false, error);
+                        }
+
+                        readErrorStringPropertyResult = await this.TryReadErrorStringPropertyValueAsync()
+                            .ConfigureAwait(false);
+                        if (!readErrorStringPropertyResult.Item1)
+                        {
+                            return Tuple.Create(false, error);
+                        }
+
+                        error.Message = readErrorStringPropertyResult.Item2;
+                        break;
+
+                    case JsonConstants.ODataErrorTargetName:
+                        if (!ODataJsonLightReaderUtils.ErrorPropertyNotFound(
+                            ref propertiesFoundBitmask,
+                            ODataJsonLightReaderUtils.ErrorPropertyBitMask.Target))
+                        {
+                            return Tuple.Create(false, error);
+                        }
+
+                        readErrorStringPropertyResult = await this.TryReadErrorStringPropertyValueAsync()
+                            .ConfigureAwait(false);
+                        if (!readErrorStringPropertyResult.Item1)
+                        {
+                            return Tuple.Create(false, error);
+                        }
+
+                        error.Target = readErrorStringPropertyResult.Item2;
+                        break;
+
+                    case JsonConstants.ODataErrorDetailsName:
+                        if (!ODataJsonLightReaderUtils.ErrorPropertyNotFound(
+                                ref propertiesFoundBitmask,
+                                ODataJsonLightReaderUtils.ErrorPropertyBitMask.Details))
+                        {
+                            return Tuple.Create(false, error);
+                        }
+
+                        Tuple<bool, List<ODataErrorDetail>> readErrorDetailsPropertyResult = await this.TryReadErrorDetailsPropertyValueAsync()

```

Comment:
Consider naming the tuple fields for better readibility.

Reply:
See my comment here https://github.com/OData/odata.net/pull/2264#discussion_r759368356